TYPO3.Flow/Classes/TYPO3/Flow/Package.php

```diff
@@ -125,6 +125,21 @@ public function boot(Core\Bootstrap $bootstrap)
             }
         });
 
+        $dispatcher->connect(\TYPO3\Flow\Monitor\FileMonitor::class, 'filesHaveChanged', function ($monitorIdentifier, $changedFiles) use ($bootstrap) {
+            if ($monitorIdentifier !== 'Flow_ClassFiles') {
+                return;
+            }
+            $packageManager = $bootstrap->getEarlyInstance(PackageManagerInterface::class);
+            $packagePhpReversed = strrev('Package.php');
+            foreach ($changedFiles as $pathAndFilename => $status) {
+                if (strpos(strrev($pathAndFilename), $packagePhpReversed) === 0) {
```

Comment:
It is basically a check for "endsWith" done a bit cryptic for performance.

i.e. `strpos(strrev($pathAndFilename), "php.egakcaP") === 0 <=> strpos($pathAndFilename, "Package.php") === strlen($pathAndFilename) - strlen("Package.php")`